sw/device/tests/closed_source/meson.build

```diff
@@ -0,0 +1,37 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# All build targets below will be built in closed source environment only. 
+# In order to build these targets, 'build_closed_source' and 'closed_source_dir'
+# build flags should be defined.
+
+closed_source_dir = get_option('closed_source_dir')
+closed_source_test_path = 'sw/device/tests/'
+
+hookfile = join_paths(closed_source_dir , closed_source_test_path, \
+                      'ip_config_hooks.c')
```

Comment:
I wonder if it might be better to retrieve the list of dependencies that are added external to open source repo as `get_option` option. That way, the compilation would live in the external meson build file and the open source has no need to assume what sources are supposed to exist.

Reply:
I also wished to exclude the file list out of the open-source. But given the meson restrictions, I can't really do that. Maybe it will be easier with Bazel. If you have a working solution for it, I'd love to see it.

Reply:
@MaayanKashani could you please elaborate why @sriyerg's suggestion here will not work? I think it will. Basically, he is saying that instead of testing for an empty "closed_source_dir" string option, you could just provide the list of closed source files as the **single** build option. This build option would be an [array type build option](https://mesonbuild.com/Build-options.html#arrays). Then, you could check the array's size to see if it is empty, i.e., we are in the open-source domain so there would be no closed source files. Does that make sense?

Reply:
It does make sense, but thinking forward, this infrastructure will serve more needs than just testing, for example, adding closed source DIFs. So, this directory is needed as an entry point of all closed source files. I can replace this flag with closed_source_test_files for example, and it will be a list type option. But, in the future, we will have another flags for each file list needed. So , i'm not sure it's the best solution for the generic infrastructure issue.

Reply:
Well, I tried converting the file list into an option-type array. But then encountered another issue- I need to have some kind of pattern for me to pair the closed source file with the relevant open-source test. I will probably use some kind of string manipulation, but for now, can we leave it as is, and just remove the build_closed_source flag? We wish to solve it in the future.

Reply:
That's ok, we can leave as is for now, and update later. Let's just fix the formatting issue stated below so we can pass CI and get this merged.